src/Crypto/Ed25519SignKey.cs

```diff
@@ -0,0 +1,83 @@
+using NSec.Cryptography;
+
+namespace Concordium.Sdk.Crypto;
+
+/// <summary>
+/// Represents a (secret) sign key of an ed25519 keypair.
+///
+/// Used for signing transactions.
+/// </summary>
+public record Ed25519SignKey : ISigner
+{
+    /// <summary>
+    /// The length of an ed25519 (secret) sign key in bytes.
+    /// </summary>
+    public const int SignKeyBytesLength = 32;
+
+    /// <summary>
+    /// Representation of the ed25519 (secret) sign key as a length-32 byte array.
+    /// </summary>
+    private readonly byte[] _value;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Ed25519SignKey"/> class.
+    /// </summary>
+    /// <param name="signKeyAsBytes">A length-32 byte array representing the ed25519 (secret) sign key.</param>
+    private Ed25519SignKey(byte[] signKeyAsBytes) => this._value = signKeyAsBytes;
+
+    /// <summary>
+    /// Gets the ed25519 (secret) sign key represented as a length-64 hex encoded string.
+    /// </summary>
+    public override string ToString() => Convert.ToHexString(this._value).ToLowerInvariant();
+
+    /// <summary>
+    /// Gets the ed25519 (secret) sign key represented as a length-32 byte array.
+    /// </summary>
+    public byte[] GetBytes() => (byte[])this._value.Clone();
```

Comment:
Why is this cloned? If I was the user of this I would expect the reference to be returned and this call is quite expense since you also need to box it with `(byte[])`

Reply:
For encapsulation and immutability of fields. I think that the interface of the sign key class, and SDK classes in general, should be agnostic to their internal representations. I would personally not make the assumption that a bytestring reference returned from this function is a reference to internal representation, since it could be anything.

I would not imagine that allocating and boxing a byte array of constant size to be a performance concern, but it is possible to use a `ReadOnlyCollection`, I suppose. It does add some abstraction for something which is quite simple. I would understand it if some Amdahl's law type of argument could be made, or if the array was large and/or dynamic as this could cause scaling issues.

Reply:
I changed the method names to `ToBytes` where applicable, used `ToArray` rather than cloning, and updated the comment doc.